.github/workflows/ci-cd.yml

```
@@ -0,0 +1,527 @@
+            Environment: ${{ env.NEXT_ENV }}
+
+  # Rollback capability
+  rollback:
+    runs-on: ubuntu-latest
+    if: failure() && github.ref == 'refs/heads/main'
+    needs: [deploy-production]
+    environment: production
+
+    steps:
+      - name: Rollback production deployment
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:7:74: Double quote to prevent globbing and word splitting

(shellcheck)